gateway/src/service.rs

```diff
@@ -515,6 +526,26 @@ impl GatewayService {
             .map_err(|_| Error::from_kind(ErrorKind::Internal))
     }
 
+    pub async fn find_custom_domain_for_project(
```

Comment:
Utility method for `project->domain` lookup. Inverse of the method below which performs `domain->project`.